test/model/internal_link_test.dart

```diff
@@ -54,6 +54,37 @@ void main() {
     }
   }
 
+  void testExpectedUrls(List<(String, String)> testCases, {
+    List<ZulipStream>? streams,
+    List<User>? users,
+  }) {
+    assert(streams != null || users != null);
+    for (final testCase in testCases) {
+      final String inputUrl = testCase.$1;
+      final String expectedUrl = testCase.$2;
```

Comment:
nit:
```suggestion
   for (final (inputUrl, expectedUrl) in testCases) {
```
I can see that this helper parallels `testExpectedNarrows`, where the same pattern is also applicable.